src/main/solution_space/path_finder_test_system.py

```diff
@@ -0,0 +1,204 @@
+            table.add_row(class_row)
+        return TestSystem.__set_table_style(table)
+
+    # Get a table with all path_finder results for a given test_inputs:
+    # +---------+-----+-------+---------------+---------------+
+    # | ... input headers...  | path_finder_1 | path_finder_2 |
+    # +---------+-----+-------+---------------+---------------+
+    # |     test_input_1      |   *result*    |   *result*    |
+    # +---------+-----+-------+---------------+---------------+
+    # |     test_input_2      |   *result*    |   *result*    |
+    # +---------+-----+-------+---------------+---------------+
+    def get_result_table(self, title: str) -> PrettyTable:
+        path_finders = self.__get_all_possible_path_finders()
+        # Set table headers: first go test_input headers, then path_finder versions
+        table = PrettyTable(field_names=[e.value for e in TEST_INPUT] +
+                                        [self.__get_path_finder_version(pf) for pf in path_finders], title=title)
+
+        for test_input in self._test_inputs:
+            user_vertex = self.__create_user_vertex(test_input)
+            row = [test_input[key] for key in TEST_INPUT]
+            for path_finder in path_finders:
+                row.append(self.__run_path_finder(path_finder, user_vertex))
+            table.add_row(row)
+
+        return TestSystem.__set_table_style(table)
+
+    def __get_all_possible_path_finders(self) -> List[IPathFinder]:
+        path_finders = []
+        for path_finder_subclass in self._path_finder_subclasses:
+            for measured_vertex_subclass in self.__get_all_subclasses(IMeasuredVertex):
+                path_finders.append(path_finder_subclass(self._graph, measured_vertex_subclass))
+        return path_finders
+
+    def __create_user_vertex(self, test_input: TestInput) -> Vertex:
+        # todo: find rate
+        rate = 0
+        vertex = Vertex(self._graph, Code.from_source(test_input[TEST_INPUT.SOURCE_CODE], rate))
+        code_info = CodeInfo(User(Profile(test_input[TEST_INPUT.AGE], test_input[TEST_INPUT.EXPERIENCE])))
+        vertex.add_code_info(code_info)
+        return vertex
+
+    @staticmethod
+    def __run_path_finder(path_finder: IPathFinder, user_vertex: Vertex) -> str:
+        # time_format = '%m/%d/%y %H:%M:%S'
+        start_time = datetime.now()
+        next_vertex = path_finder.find_next_vertex(user_vertex)
+        end_time = datetime.now()
+        return f'time: {end_time - start_time}\n\n' \
+               f'vertex id: {next_vertex.id}\n\n' \
+               f'canon code:\n{get_code_from_tree(next_vertex.canon_tree)}'
+
+    # Gets path_finder version in format like this: 'PathFinderV1, MeasuredVertexV1'
+    @staticmethod
+    def __get_path_finder_version(path_finder: IPathFinder) -> str:
+        return f'{type(path_finder).__name__},{path_finder.measured_vertex_subclass.__name__}'
+
+    @staticmethod
+    def __format_doc_str(doc: str) -> str:
+        spaces_to_crop = 8
```

Comment:
You have a lot of const variables like this. I think we can move it to a const file or to the top of the document

Reply:
Yeah, you rights